### DISCUSSION TITLE
API /inventory endpoint 'no implicit conversion of Symbol into Integer' error

### DISCUSSION THREAD

**Discussion by emoty88** [Sellbrite/Sellbrite-API#7] [Q&A]
I've added some products through the product API and tried to add inventory to the product, but all my request to /inventory endpoint returns the same error. 

The response is 
`{'error': {'message': 'There was an error with your request. (no implicit conversion of Symbol into Integer)'}}`

I've sent post with the body;
```
{
   "inventory":{
      "sku":"FAKEASIN-11",
      "warehouse_uuid":"01700f43-7e6b-44da-99af-3179382b6dcb",
      "on_hand":1
   }
}
```

i've sent the patch and also put requests with the body of json those I provided below;

```
{
   "inventory":{
      "sku":"FAKEASIN-11",
      "description":"Initial Stock",
      "warehouse_uuid":"01700f43-7e6b-44da-99af-3179382b6dcb",
      "available":1,
      "on_hand":1,
      "bin_location":"100",
      "cost":10
   }
}
```
--
```
{
   "inventory":{
      "sku":"FAKEASIN-11",
      "description":"Initial Stock",
      "warehouse_uuid":"01700f43-7e6b-44da-99af-3179382b6dcb",
      "on_hand":1
   }
}
```

All my requests get same response ;
`{'error': {'message': 'There was an error with your request. (no implicit conversion of Symbol into Integer)'}}`

Please Help me with this issue.

**Comment by justinmcdaniel** [comment]
@emoty88 I have a PUT call to this endpoint working in my systems. I have not tried the POST call, so I cannot speak for it, but the PUT and PATCH calls can take up to 50 records. With the PUT call, even if I only send 1 record, I have to wrap the "inventory" param as an array, even though the example shows it without an array. I believe the old documentation did show the array brackets. Try that and maybe that will fix it.

i.e.

Instead of:

```json
{
   "inventory":{
      "sku":"FAKEASIN-11",
      "description":"Initial Stock",
      "warehouse_uuid":"01700f43-7e6b-44da-99af-3179382b6dcb",
      "on_hand":1
   }
}
```

try: 
```json
{
   "inventory": [{
      "sku":"FAKEASIN-11",
      "description":"Initial Stock",
      "warehouse_uuid":"01700f43-7e6b-44da-99af-3179382b6dcb",
      "on_hand":1
   }]
}
```
